### PR TITLE
Dev docs standardization of format, style, and language, plus other minor changes and edits. 

### PR DESCRIPTION
**Pull request by jessiemongeon1**
Thank you for your contribution to the IC Developer Portal.
Before submitting your Pull Request, please make sure that:

- [x] You have updated the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file according to your changes.
- [x] You have registered new documents to [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js) if any.

This PR standardizes the Dev Docs pages within sections 'Tutorials' and 'Guides' to reflect the following:
- A cohesive, uniform format across all pages, and fixed errors such as typos, broken links, and incorrectly formatted text. This includes fixing capitalization, punctuation, use of bullet points, bold text, and italic text, and different heading sizes.
- Edited the sidebar to assure that page titles matched sidebar titles and included links to additional relevant pages. 
- Links to existing Sample Code tutorial pages as hyperlinked subpages in the 'Tutorials' section.
- Links the Concepts documentation pages as hyperlinked subpages for the ‘Introduction’ section of the 'Guides' category, since they provide introductory information into the IC’s concepts. 
- Links pages from the Tokenomics documentation page through hyperlinked subpages that contain important guides that aren’t referenced in the Dev Docs, such as the NNS Dapp Quick Start and the Self Custody Quick Start. 
- Moved the ‘SNS Introduction’ page that currently sits by itself at the bottom of the SNS section into the ‘SNS & DAO Primer’ section. 
- Moved the 'Gas / Cycles' page into the 'Introduction' section. 
- Reordered the ‘Building Smart Contracts with Motoko’ subpages order.
- Sorted subsections by alphabetical order, unless a different logical order was a better fit. 
- Renamed certain pages for clarity and uniform format. 
- Standardized the use of currency codes to use XDR in place of SDR.

**This PR covers the 'Tutorials' and 'Guides' categories. The 'References' category will be next, followed by the addition of the Style Guide that provides details for each standardized change.**

### CODE REVIEW

File: docs/developer-docs/setup/install/index.mdx [hunk 1]
```diff
@@ -157,25 +159,27 @@ The `~/.cache/dfinity/versions` directory stores one or more versioned subdirect
...
+You don't need to uninstall the software before installing the new version. However, if you want to perform a clean installation rather than an upgrade, you can first uninstall the software as described in [removing the software](#remove), then re-run the download and installation command.
+:::
 
-For information about the features and fixes in the latest release, see the [Release notes](/other/updates/release-notes/release-notes.md).
+For information about the features and fixes in the latest release, see the [release notes](/other/updates/release-notes/release-notes.md).
 
 ## Installing a specific version
 
-The install script looks for the environment variable `DFX_VERSION` to check if it is supposed to install a specific version of the SDK or to simply install the latest version.
+The install script looks for the environment variable `DFX_VERSION` to check if it is supposed to install a specific version of the ICSDK or to simply install the latest version.
```
Comment (dprats) [inline docs/developer-docs/setup/install/index.mdx:182]: change to IC SDK

**Review by dprats — CHANGES_REQUESTED** [review]
I left some small comments. Should be easy to change.

**Review by dprats — APPROVED** [review]

**Comment by LaraAS** [comment]
I see the comment 
> Moved the ‘SNS Introduction’ page that currently sits by itself at the bottom of the SNS section into the ‘SNS & DAO Primer’ section.
To clarify: I think this page should be after the tokenomics but before the get-sns pages (As outlined in [this document](https://docs.google.com/document/d/147FZK9KhF4rPQu3wJSwiPV5sHebjjJOi30xDtj3dIgg/edit)).
I am not sure how it got moved to where it is now. 
Would this be a valuable option?

(I write here because I am unsure in which file this change is specified.)

File: docs/developer-docs/integrations/rosetta/index.md [hunk 2]
```diff
@@ -8,68 +10,67 @@ This documentation might help if you want to deploy a token on the IC that aims
...
+```
 
-    :::
+:::info
+The first time you run the command, it might take some time for the node to catch up to the current link of the chain.
+When the node is caught up, you should see output similar to the following:
+:::
 
-        You are all caught up to block height 109
+    You are all caught up to block height 109
```
Comment (roman-kashitsyn) [inline docs/developer-docs/integrations/rosetta/index.md:65]: I believe this message belongs the "info" block above.

**Review by roman-kashitsyn — APPROVED** [review]
Monumental work, thank you!

File: docs/developer-docs/integrations/rosetta/index.md (hunk 2 above)
Comment (jessiemongeon1) [reply docs/developer-docs/integrations/rosetta/index.md]: Good catch, thank you!

File: docs/developer-docs/security/rust-canister-development-security-best-practices.md [hunk 3]
```diff
@@ -419,19 +428,19 @@ If a canister traps or panics in `pre_upgrade`, this can lead to permanently blo
...
+- See [current limitations of the Internet Computer](https://wiki.internetcomputer.org/wiki/Current_limitations_of_the_Internet_Computer), section "Bugs in `pre_upgrade` hooks".
 
 ### Reinstantiate timers during upgrades
 
 #### Security Concern
 
 Global timers are deactivated upon changes to the canister's Wasm module. The [IC specification](../../references/ic-interface-spec#timer) states this as follows:
 
-> The timer is also deactivated upon changes to the canister's Wasm module (calling install_code, uninstall_code methods of the management canister or if the canister runs out of cycles). In particular, the function canister_global_timer won't be scheduled again unless the canister sets the global timer again (using the System API function ic0.global_timer_set).
+The timer is also deactivated upon changes to the canister's Wasm module (calling install_code, uninstall_code methods of the management canister or if the canister runs out of cycles). In particular, the function canister_global_timer won't be scheduled again unless the canister sets the global timer again (using the System API function ic0.global_timer_set).
```
Comment (robin-kunzler) [inline docs/developer-docs/security/rust-canister-development-security-best-practices.md:443]: Only a minor thing: Here I'm quoting from the IC specification and thus use the Markdown Blockquote ">". Did you intentionally remove it to unify with other places? Or is there maybe still some way of showing this paragraph as a quote?

**Review by robin-kunzler — APPROVED** [review]
Thanks a lot @jessiemongeon1 for this standardization effort. This looks great!

I only have one minor comment/question, see below.

File: docs/developer-docs/security/rust-canister-development-security-best-practices.md (hunk 3 above)
Comment (jessiemongeon1) [reply docs/developer-docs/security/rust-canister-development-security-best-practices.md]: I've pushed a change to reinstate the Blockquote with ">" and added quotation marks around the quote to further emphasis that this is a direct quote from the IC spec page.

**Pull request merged**